src/evm_builtins.rs

```diff
@@ -20,109 +20,99 @@ impl Instructions for EVMInstructions {
         arguments: Vec<SMTVariable>,
         return_vars: &[SMTVariable],
         ssa: &mut SSATracker,
-    ) -> String {
-        let single_return = |value: String| {
+    ) -> SMTStatement {
+        let single_return = |value: SMTExpr| {
             assert_eq!(return_vars.len(), 1);
-            format!(
-                "(define-const {} (_ BitVec 256) {})",
-                &return_vars.first().unwrap().name,
-                value
-            )
+            smt::define_const(return_vars.first().unwrap().clone(), value)
         };
-        let direct = |smt_name: &str| {
-            let smt_encoding =
-                format!("({} {} {})", smt_name, arguments[0].name, arguments[1].name);
-            single_return(match is_bool_function(smt_name) {
+        let direct = |smt_op: SMTOp| {
+            let smt_encoding = SMTExpr {
+                op: smt_op.clone(),
+                args: arguments.clone().into_iter().map(|v| v.into()).collect(),
+            };
+            single_return(match is_bool_function(smt_op) {
                 true => wrap_boolean(smt_encoding),
                 false => smt_encoding,
             })
         };
 
+        let mut it = arguments.clone().into_iter();
+        let arg_0 = it.next();
+        let arg_1 = it.next();
```

Comment:
```suggestion
        let arg_0 = arguments.nth(0);
        let arg_1 = arguments.nth(1);
```
?

Reply:
that way we'd need to clone it after, cus `iter` doesn't. The previous option clones and moves when used